metricbeat/module/kubernetes/container/container.go

```diff
@@ -78,31 +78,30 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 // Fetch methods implements the data gathering and data conversion to the right
 // format. It publishes the event which is then forwarded to the output. In case
 // of an error set the Error field of mb.Event or simply call report.Error().
-func (m *MetricSet) Fetch(reporter mb.ReporterV2) {
+func (m *MetricSet) Fetch(reporter mb.ReporterV2) error {
 	m.enricher.Start()
 
 	body, err := m.http.FetchContent()
 	if err != nil {
-		err = errors.Wrap(err, "error doing HTTP request to fetch 'container' Metricset data")
-		logger.Error(err)
-		reporter.Error(err)
-		return
+		return errors.Wrap(err, "error doing HTTP request to fetch 'container' Metricset data")
+
 	}
 
 	events, err := eventMapping(body, util.PerfMetrics)
 	if err != nil {
-		logger.Error(err)
-		reporter.Error(err)
-		return
+		return errors.Wrap(err, "error in mapping")
 	}
 
 	m.enricher.Enrich(events)
 
 	for _, e := range events {
-		reporter.Event(mb.TransformMapStrToEvent("kubernetes", e, nil))
+		isOpen := reporter.Event(mb.TransformMapStrToEvent("kubernetes", e, nil))
```

Comment:
Is this when the metricset is stopped?

Reply:
`Event` will return false when the metricset reporter stops, if that's what you're asking.

Reply:
yes. The variable name confused me here but I don't have much better proposals at the moment.